From 3e046fc33b54a331ddd2901d48c34cb753873c2a Mon Sep 17 00:00:00 2001
From: masemoel <az.az.az.mjme@gmail.com>
Date: Mon, 21 Dec 2020 18:46:00 +0100
Subject: [PATCH 1/2] Specify heap size for metalava for R

filiprrs: This is needed on systems with 8GB physical ram.
Compiling using a single job is recommended.

Co-authored-by: althafvly <althafvly@gmail.com>
Co-authored-by: nparashar150 <nparashar150@gmail.com>
Change-Id: I5093dcdbe384fd33f0dc0fd9b89c91f8f9fa19fd
---
 java/droiddoc.go | 1 +
 1 file changed, 1 insertion(+)

diff --git a/java/droiddoc.go b/java/droiddoc.go
index b564fea0..cdaf8a6c 100644
--- a/java/droiddoc.go
+++ b/java/droiddoc.go
@@ -1474,6 +1474,7 @@ func metalavaCmd(ctx android.ModuleContext, rule *android.RuleBuilder, javaVersi
 
 	cmd.BuiltTool(ctx, "metalava").
 		Flag(config.JavacVmFlags).
+		Flag("-J-Xmx3112m").
 		FlagWithArg("-encoding ", "UTF-8").
 		FlagWithArg("-source ", javaVersion.String()).
 		FlagWithRspFileInputList("@", srcs).
-- 
2.33.0

